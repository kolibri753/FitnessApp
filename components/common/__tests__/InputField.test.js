import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import InputField from "../InputField";

describe("InputField", () => {
	it("renders correctly with default props", () => {
		const { getByPlaceholderText } = render(<InputField />);
		expect(getByPlaceholderText("Enter text...")).toBeTruthy();
	});

	it("renders correctly with provided props", () => {
		const { getByPlaceholderText } = render(
			<InputField
				placeholder="Custom placeholder"
				value="Test value"
				onChangeText={() => {}}
				secureTextEntry={true}
				error="Test error message"
				multiline={true}
				numberOfLines={4}
			/>
		);
		expect(getByPlaceholderText("Custom placeholder")).toBeTruthy();
	});

	it("calls onChangeText function when input value changes", () => {
		const onChangeTextMock = jest.fn();
		const { getByTestId } = render(
			<InputField onChangeText={onChangeTextMock} />
		);
		fireEvent.changeText(getByTestId("inputField"), "New value");
		expect(onChangeTextMock).toHaveBeenCalledWith("New value");
	});

	it("toggles secureTextEntry when icon is pressed", () => {
		const { getByTestId } = render(
			<InputField
				placeholder="Password"
				value="Test password"
				secureTextEntry={true}
			/>
		);
		const eyeIcon = getByTestId("eyeIcon");
		fireEvent.press(eyeIcon);
		expect(getByTestId("eyeIcon")).toBeTruthy();
	});

	it("displays error message when error prop is provided", () => {
		const { getByText } = render(
			<InputField placeholder="Text" error="Test error message" />
		);
		expect(getByText("Test error message")).toBeTruthy();
	});

  it("renders correctly", () => {
		const tree = renderer.create(<InputField />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
