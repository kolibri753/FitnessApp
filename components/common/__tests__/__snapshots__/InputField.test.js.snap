// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <TextInput
    placeholder="Enter text..."
    placeholderTextColor="#FFFFFF"
    selectionColor="#FFD500"
    style={
      [
        {
          "backgroundColor": "#000000",
          "borderColor": "#FFD500",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#FFFFFF",
          "flex": 1,
          "marginBottom": 20,
          "padding": 10,
          "position": "relative",
        },
        null,
        null,
      ]
    }
    testID="inputField"
    textAlignVertical="center"
  />
</View>
`;
